File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples-js/Diagnostics_ListHostingEnvironmentDetectorResponses.js
const { WebSiteManagementClient } = require("@azure/arm-appservice");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Description for List Hosting Environment Detector Responses
 *
 * @summary Description for List Hosting Environment Detector Responses
 * x-ms-original-file: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples/Diagnostics_ListHostingEnvironmentDetectorResponses.json
 */
async function getAppServiceEnvironmentDetectorResponses() {
  const subscriptionId = "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345";
  const resourceGroupName = "Sample-WestUSResourceGroup";
  const name = "SampleAppServiceEnvironment";
  const credential = new DefaultAzureCredential();
  const client = new WebSiteManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.diagnostics.listHostingEnvironmentDetectorResponses(
    resourceGroupName,
    name
  )) {
    resArray.push(item);
  }
  console.log(resArray);
}

getAppServiceEnvironmentDetectorResponses().catch(console.error);

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples-java/Diagnostics_ListHostingEnvironmentDetectorResponses.java
import com.azure.core.util.Context;

/** Samples for Diagnostics ListHostingEnvironmentDetectorResponses. */
public final class Main {
    /*
     * x-ms-original-file: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples/Diagnostics_ListHostingEnvironmentDetectorResponses.json
     */
    /**
     * Sample code: Get App Service Environment Detector Responses.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void getAppServiceEnvironmentDetectorResponses(com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .webApps()
            .manager()
            .serviceClient()
            .getDiagnostics()
            .listHostingEnvironmentDetectorResponses(
                "Sample-WestUSResourceGroup", "SampleAppServiceEnvironment", Context.NONE);
    }
}
